trio/_core/_traps.py

```diff
@@ -64,7 +65,10 @@ class WaitTaskRescheduled:
     abort_func = attr.ib()
 
 
-async def wait_task_rescheduled(abort_func):
+RaiseCancelT = Callable[[], NoReturn]  # TypeAlias
```

Comment:
I'm not sure about this naming or having a type alias here. But it's your call! The type alias does seem useful, even if I'm not the biggest fan of the name.